cmd/traffic/cmd/manager/internal/mutator/service.go

```diff
@@ -0,0 +1,182 @@
+package mutator
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	admission "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+
+	"github.com/datawire/dlib/dlog"
+	"github.com/datawire/dlib/dutil"
+	"github.com/telepresenceio/telepresence/v2/pkg/install"
+)
+
+const (
+	tlsDir          = `/var/run/secrets/tls`
+	tlsCertFile     = `crt.pem`
+	tlsKeyFile      = `key.pem`
+	jsonContentType = `application/json`
+)
+
+var universalDeserializer = serializer.NewCodecFactory(runtime.NewScheme()).UniversalDeserializer()
+
+// JSON patch, see https://tools.ietf.org/html/rfc6902 .
+type patchOperation struct {
+	Op    string      `json:"op"`
+	Path  string      `json:"path"`
+	Value interface{} `json:"value,omitempty"`
+}
+
+type mutatorFunc func(context.Context, *admission.AdmissionRequest) ([]patchOperation, error)
+
+func ServeMutator(ctx context.Context) error {
+	certPath := filepath.Join(tlsDir, tlsCertFile)
+	keyPath := filepath.Join(tlsDir, tlsKeyFile)
+	missing := ""
+	if _, err := os.Stat(certPath); err != nil {
+		if !os.IsNotExist(err) {
+			return err
+		}
+		missing = certPath
+	} else if _, err = os.Stat(keyPath); err != nil {
+		if !os.IsNotExist(err) {
+			return err
+		}
+		missing = keyPath
+	}
+	if missing != "" {
+		dlog.Infof(ctx, "%q is not present so mutator service is disabled", missing)
+		return nil
+	}
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/traffic-agent", func(w http.ResponseWriter, r *http.Request) {
+		dlog.Info(ctx, "Received webhook request...")
+		bytes, statusCode, err := serveMutatingFunc(ctx, r, agentInjector)
+		if err != nil {
+			dlog.Errorf(ctx, "error handling webhook request: %v", err)
+			w.WriteHeader(statusCode)
+			bytes = []byte(err.Error())
+		} else {
+			dlog.Info(ctx, "Webhook request handled successfully")
+		}
+		if _, err = w.Write(bytes); err != nil {
+			dlog.Errorf(ctx, "could not write response: %v", err)
+		}
+	})
+	mux.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	})
+
+	server := &http.Server{
+		Addr:     ":" + strconv.Itoa(install.MutatorWebhookPortHTTPS),
+		ErrorLog: dlog.StdLogger(ctx, dlog.LogLevelError),
+		Handler:  mux}
+
+	dlog.Infof(ctx, "Mutating webhook service is listening on %v", server.Addr)
+	err := dutil.ListenAndServeHTTPSWithContext(ctx, server, certPath, keyPath)
```

Comment:
If you use `dhttp` instead of the old deprecated `dutil`, then you don't have to configure `ErrorLog`.